src/pages/viewer-app/viewer-app.tsx

```diff
@@ -238,8 +248,9 @@ export const ViewerApp = () => {
     setSelectedFeatureAttributes(null);
     setSelectedFeatureIndex(-1);
     if (buildingExplorerOpened && tilesetsData[0]) {
+      const tilesetDataUrl = typeof tilesetsData[0].url === "string" ? tilesetsData[0].url : tilesetsData[0].url.name;
```

Comment:
Some utility `function getDatasetUrl(dataset: LayerExample | TilesetMetadata) => string`. You reuse this pattern many times. This function would make the code more readable.

```suggestion
      const tilesetDataUrl = getDatasetUrl(tilesetsData[0]);
```